src/coreclr/System.Private.CoreLib/src/Internal/Runtime/Binder/ApplicationContext.cs

```diff
@@ -0,0 +1,309 @@
+
+        public Dictionary<FailureCacheKey, int> FailureCache { get; } = new Dictionary<FailureCacheKey, int>();
+
+        public object ContextCriticalSection { get; } = new object();
+
+        private readonly List<string> _platformResourceRoots = new List<string>();
+
+        public List<string> AppPaths { get; } = new List<string>();
+
+        public Dictionary<string, TPAEntry>? TrustedPlatformAssemblyMap { get; private set; }
+
+        public void IncrementVersion() => Interlocked.Increment(ref _version);
+
+        private const char PATH_SEPARATOR_CHAR = ';';
+
+        private static bool GetNextPath(string paths, ref int startPos, out string outPath)
+        {
+            bool wrappedWithQuotes = false;
+
+            // Skip any leading spaces or path separators
+            while (startPos < paths.Length && paths[startPos] is ' ' or PATH_SEPARATOR_CHAR)
+                startPos++;
+
+            if (startPos == paths.Length)
+            {
+                // No more paths in the string and we just skipped over some white space
+                outPath = string.Empty;
+                return false;
+            }
+
+            // Support paths being wrapped with quotations
+            while (startPos < paths.Length && paths[startPos] == '\"')
+            {
+                startPos++;
+                wrappedWithQuotes = true;
+            }
+
+            int iEnd = startPos; // Where current path ends
+            int iNext;           // Where next path starts
+
+            if (wrappedWithQuotes)
+            {
+                iEnd = paths.AsSpan(iEnd).IndexOf('\"');
+                if (iEnd != -1)
+                {
+                    // Find where the next path starts - there should be a path separator right after the closing quotation mark
+                    iNext = paths.AsSpan(iEnd).IndexOf(PATH_SEPARATOR_CHAR);
+                    if (iNext != -1)
+                    {
+                        iNext++;
+                    }
+                    else
+                    {
+                        iNext = paths.Length;
+                    }
+                }
+                else
+                {
+                    // There was no terminating quotation mark - that's bad
+                    throw new ArgumentException(nameof(paths));
+                }
+            }
+            else if ((iEnd = paths.AsSpan(iEnd).IndexOf(PATH_SEPARATOR_CHAR)) != -1)
+            {
+                iNext = iEnd + 1;
+            }
+            else
+            {
+                iNext = iEnd = paths.Length;
+            }
+
+            // Skip any trailing spaces
+            while (paths[iEnd - 1] == ' ')
+            {
+                iEnd--;
+            }
+
+            Debug.Assert(startPos < iEnd);
+
+            outPath = paths[startPos..iEnd];
+            startPos = iNext;
+            return true;
+        }
+
+        private static bool IsRelativePath(ReadOnlySpan<char> path)
```

Comment:
It should be fine to replace this by `Path.IsPathFullyQualified`.

Reply:
I noticed the behavior difference with `IsPathRooted`, but didn't mention this.

Reply:
The corner-case behavior differences should not be a problem.